perceval/algorithm/tomography/tomography_mle.py

```diff
@@ -0,0 +1,400 @@
+        # Gives a set of measurement states, so they form a set of informationally complete measurements.
+        # For 1 qubit, they are (order important) : |0>,|1>,|+>,|->,|i+>,|i->,
+        # These measurement states are eigenvectors of the tensor products of Pauli operators
+
+        d = 2 ** self._nqubit
+        P = []
+        B = [np.array([[1], [0]], dtype='complex_'), np.array([[0], [1]], dtype='complex_'),
+             (1 / np.sqrt(2)) * np.array([[1], [1]], dtype='complex_'),
+             (1 / np.sqrt(2)) * np.array([[1], [-1]], dtype='complex_'),
+             (1 / np.sqrt(2)) * np.array([[1], [1j]], dtype='complex_'),
+             (1 / np.sqrt(2)) * np.array([[1], [-1j]], dtype='complex_')]  #
+
+        for pauli_index in range(3 ** self._nqubit):
+            # iterates over pauli operators I(equivalent to Z),X,Y or their tensor products
+            pauli_list = []  # saves on each element the pauli index for this qubit
+
+            for j in range(self._nqubit - 1, -1, -1):
+                pauli_list.append(pauli_index // (3 ** j))
+                pauli_index = pauli_index % (3 ** j)
+            X = [np.array([1], dtype='complex_')] * d  # neutral elements for tensor products
+
+            for qubit_index in range(self._nqubit):
+                for eigenvector_index in range(d):  # iterates over the eigenvectors of the pauli operator
+                    eigenvector_list = []  # saves on each element the eigenvector index for this qubit
+                    k1 = eigenvector_index
+                    for j1 in range(self._nqubit - 1, -1, -1):
+                        eigenvector_list.append(k1 // (2 ** j1))
+                        k1 = k1 % (2 ** j1)
+                    X[eigenvector_index] = np.kron(X[eigenvector_index],
+                                                   B[2 * pauli_list[qubit_index] + eigenvector_list[qubit_index]])
+            P += X
+        return P
+
+    def _povm_operator(self) -> list:
+        # Gives a POVM (positive operator value measure) set suited for tomography
+        B = self._povm_state()
+        L = []
+        for state in B:
+            L.append(_state_to_dens_matrix(state) / (3 ** self._nqubit))
+        return L
+
+    def _compute_matrix(self, j: int) -> np.ndarray:
+        if j == 0:
+            return np.eye((2), dtype='complex_')
+        if j == 1:
+            return (1 / np.sqrt(2)) * np.array([[1, 1], [1, -1]], dtype='complex_')
+        if j == 2:
+            return (1 / np.sqrt(2)) * np.array([[1, 1], [1j, -1j]], dtype='complex_')
+
+    def _input_basis(self) -> list:
+        # Computes input density matrix basis (similar to POVM but not same order)
+
+        B = []
+        for j in range(6 ** self._nqubit):
+            k = []
+            for m in range(self._nqubit - 1, -1, -1):
+                k.append(j // (6 ** m))
+                j = j % (6 ** m)
+            k.reverse()
+            M = 1
+            v = 1
+            for i in k:
```

Comment:
instead of reversing k you can iterate over the reverse vector:
`for i in reversed(k):`
and if you implement my version you can even do
`for i in reversed(number_to_base(j, self._nqubit, 6)):`